pkgs/applications/audio/soundux/default.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib, stdenv, fetchFromGitHub, cmake, ninja, pkg-config, makeWrapper, wrapGAppsHook,
+  libappindicator-gtk3, openssl, pipewire, pulseaudio, webkitgtk, xorg,
+  libpulseaudio, libwnck3,
+  downloaderSupport ? true, ffmpeg, youtube-dl
+}:
+
+let
+  downloaderPath = lib.makeBinPath [ffmpeg youtube-dl ];
+  # Soundux loads pipewire, pulse and libwnck optionally during runtime
+  dynamicLibraries = lib.makeLibraryPath [libpulseaudio pipewire libwnck3 ];
+in
+stdenv.mkDerivation rec {
+  pname = "soundux";
+  version = "0.2.7";
+
+  src = fetchFromGitHub {
+    owner = "Soundux";
+    repo = "Soundux";
+    rev = version;
+    fetchSubmodules = true;
```

Comment:
Do we have any of the deps there already in nixpkgs? If so they should be devondered.

Reply:
What about the deps we don't have, should those also be devendored?

Reply:
It depends. If they are libs which can be packaged separate then they should be packaged. You can do that in this PR. If they are just a git repo with some files which are not used by anything else or don't really have a build system or are not indented to be packaged separate then not.